File: nextjs-app/lib/models/role.ts
/**
 * Перечисление ролей администраторов
 * Определяет иерархию доступа в системе
 */

export enum AdminRole {
  /** Разработчик - полный доступ к системе */
  DEVELOPER = 0,
  
  /** Помощник - доступ эквивалентный разработчику */
  ASSISTANT = 1,
  
  /** Администратор - право назначать операторов */
  ADMINISTRATOR = 2,
  
  /** Оператор - базовый уровень доступа */
  OPERATOR = 3,
}

/**
 * Получает русское название роли
 * @param role - Числовое значение роли
 * @returns Название роли на русском языке
 */
export function getRoleName(role: AdminRole): string {
  switch (role) {
    case AdminRole.DEVELOPER:
      return 'Разработчик';
    case AdminRole.ASSISTANT:
      return 'Помощник';
    case AdminRole.ADMINISTRATOR:
      return 'Администратор';
    case AdminRole.OPERATOR:
      return 'Оператор';
    default:
      return 'Неизвестная роль';
  }
}

/**
 * Проверяет, может ли роль назначать операторов
 * @param role - Числовое значение роли
 * @returns true если роль <= 2 (Developer, Assistant, Administrator)
 */
export function canAssignOperators(role: AdminRole): boolean {
  return role <= AdminRole.ADMINISTRATOR;
}

/**
 * Проверяет, может ли роль изменять конфигурацию системы
 * @param role - Числовое значение роли
 * @returns true если роль <= 1 (Developer, Assistant)
 */
export function canModifyConfig(role: AdminRole): boolean {
  return role <= AdminRole.ASSISTANT;
}

/**
 * Проверяет, может ли роль отвечать пользователям
 * @param role - Числовое значение роли
 * @returns true для всех ролей (0-3)
 */
export function canRespondToUsers(role: AdminRole): boolean {
  return role >= AdminRole.DEVELOPER && role <= AdminRole.OPERATOR;
}

File: telegram-bot/models/role.py
"""
Перечисление ролей администраторов
"""

from enum import IntEnum


class AdminRole(IntEnum):
    """
    Роли администраторов системы
    
    Иерархия ролей (от высшего к низшему):
    - DEVELOPER (0): Полный доступ к системе
    - ASSISTANT (1): Доступ эквивалентный Developer
    - ADMINISTRATOR (2): Может назначать операторов
    - OPERATOR (3): Базовый уровень доступа
    
    Validates: Requirements 2.1, 2.2, 2.3, 2.4
    """
    DEVELOPER = 0
    ASSISTANT = 1
    ADMINISTRATOR = 2
    OPERATOR = 3
    
    def get_display_name(self) -> str:
        """
        Возвращает русское название роли
        
        Returns:
            Строка с названием роли на русском языке
        
        Examples:
            >>> AdminRole.DEVELOPER.get_display_name()
            'Разработчик'
            >>> AdminRole.OPERATOR.get_display_name()
            'Оператор'
        """
        role_names = {
            AdminRole.DEVELOPER: 'Разработчик',
            AdminRole.ASSISTANT: 'Помощник',
            AdminRole.ADMINISTRATOR: 'Администратор',
            AdminRole.OPERATOR: 'Оператор'
        }
        return role_names[self]
